node/src/main/scala/org/bitcoins/node/networking/peer/DataMessageHandler.scala

```diff
@@ -318,18 +313,23 @@ case class DataMessageHandler(
 object DataMessageHandler {
 
   /** Callback for handling a received block */
-  type OnBlockReceived = Block => Unit
+  type OnBlockReceived = Block => Future[Unit]
 
   /** Callback for handling a received Merkle block with its corresponding TXs */
-  type OnMerkleBlockReceived = (MerkleBlock, Vector[Transaction]) => Unit
+  type OnMerkleBlockReceived =
+    (MerkleBlock, Vector[Transaction]) => Future[Unit]
 
   /** Callback for handling a received transaction */
-  type OnTxReceived = Transaction => Unit
+  type OnTxReceived = Transaction => Future[Unit]
 
   /** Callback for handling a received compact block filter */
-  type OnCompactFilterReceived = (DoubleSha256Digest, GolombFilter) => Unit
+  type OnCompactFilterReceived =
+    (DoubleSha256Digest, GolombFilter) => Future[Unit]
+
+  /** Callback for handling a received block header */
+  type OnBlockHeadersReceived = Vector[BlockHeader] => Future[Unit]
 
   /** Does nothing */
-  def noop[T]: T => Unit = _ => ()
+  def noop[T]: T => Future[Unit] = _ => Future.successful(())
```

Comment:
nit: `FutureUtil.unit`

Reply:
`Future.unit` isn't available for scala 2.11.12